packages/editor/src/lib/editor/Editor.ts

```diff
@@ -7873,6 +7876,55 @@ export class Editor extends EventEmitter<TLEventMap> {
 		return this
 	}
 
+	/**
+	 * Register a temporary preview of an asset. This is useful for showing a ghost image of
+	 * something that is being uploaded. Returns an `RC` of the URL - call `.retain` to keep it in
+	 * memory, and `.release` when you're done with it.
+	 *
+	 * @example
+	 * ```ts
+	 * editor.setTemporaryAssetPreview('someid', file)
+	 * ```
+	 *
+	 * @param assetId - The asset's id.
+	 * @param file - The raw file.
+	 *
+	 * @public
+	 */
+	createTemporaryAssetPreview(assetId: TLAssetId, file: File) {
+		assert(!this.temporaryAssetPreview.has(assetId), 'Asset preview already exists')
```

Comment:
@SomeHats i changed this to not be an assert, because you can upload some images, delete them, and re-upload them, and then this throws